### PR TITLE
INSTUI-3352 - deprecate warning text

### PR DESCRIPTION
**Pull request by ke1k0**

### CODE REVIEW

File: docs/guides/v9-upgrade-guide.md [hunk 1]
```diff
@@ -12,19 +12,53 @@ Please read this guide carefully if you plan to update your app to this release.
 
 We recommend upgrading your application for each major version gradually, e.g. if you plan to update from 7.x to 9.x, then migrate first to 8.x and then to version 9.
 
-### Removal of deprecated packages and components
+The tables below show what will be removed and what are they replaced with. We also marked if there are [codemods](#ui-codemods) available.
 
-The following deprecated components are removed in v9:
+##### Table of Contents
```
Comment (ke1k0) [inline docs/guides/v9-upgrade-guide.md:17]: Merged the 2 guides like we agreed before (with v8 deprecations guide). Also "pimped it up" a bit and started a "deprecated prop values" block as well.

File: packages/ui-buttons/src/BaseButton/props.ts [hunk 2]
```diff
@@ -152,13 +152,12 @@ type BaseButtonOwnProps = {
...
 
   // Deprecated `string` tabIndex type
-  // TODO: remove string type in v9
   /**
    * Specifies the tabindex of the `Button`.
    *
    * (`string` type is __deprecated__, use `number`)
    */
-  tabIndex?: number | string
+  tabIndex?: number | string // TODO: remove string type in v9
```
Comment (ke1k0) [inline packages/ui-buttons/src/BaseButton/props.ts:160]: I only did these 2 changes so I can add a 'feat' commit, forgot to include in the last release.

File: packages/ui-text/src/Text/__tests__/theme.test.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+import { expect } from '@instructure/ui-test-utils'
+import { contrast } from '@instructure/ui-color-utils'
+import { canvas, canvasHighContrast, instructure } from '@instructure/ui-themes'
+
+import generateComponentTheme from '../theme'
+
+import type { BaseTheme } from '@instructure/ui-themes'
+
+// Text has to have 4.5:1 contrast in all themes
+const runTextColorContrastTest = (theme: BaseTheme) => {
```
Comment (ke1k0) [inline packages/ui-text/src/Text/__tests__/theme.test.ts:34]: Some extra tests never hurt...

File: docs/guides/v9-upgrade-guide.md (hunk 1 above)
Comment (matyasf) [reply docs/guides/v9-upgrade-guide.md:17]: Nice, I like it

**Review by matyasf — APPROVED** [review]

**Review by Brailor — APPROVED** [review]

**Pull request merged**